schema/mysql/upgrades/1.0.0-rc2.sql

```diff
@@ -322,3 +322,16 @@ ALTER TABLE flapping_history
     MODIFY id binary(20) NOT NULL COMMENT 'sha1(environment.id + "Host"|"Service" + host|service.name + start_time)';
 ALTER TABLE acknowledgement_history
     MODIFY id binary(20) NOT NULL COMMENT 'sha1(environment.id + "Host"|"Service" + host|service.name + set_time)';
+
+/*
+ * Schema changes after https://github.com/Icinga/icingadb/pull/403:
+ */
+
+ALTER TABLE downtime
+    ADD COLUMN scheduled_duration bigint unsigned NOT NULL AFTER scheduled_end_time,
+    ADD COLUMN duration bigint unsigned NOT NULL COMMENT 'Duration of the downtime: When the downtime is flexible, this is the same as flexible_duration otherwise scheduled_duration' AFTER end_time,
+    MODIFY COLUMN flexible_duration bigint unsigned NOT NULL AFTER is_flexible;
+UPDATE downtime SET scheduled_duration = scheduled_end_time - scheduled_start_time, duration = (CASE WHEN is_flexible = 'Y' THEN flexible_duration ELSE scheduled_end_time - scheduled_start_time END) WHERE scheduled_duration IS NULL;
+
+ALTER TABLE service_state ADD COLUMN host_id binary(20) NOT NULL COMMENT 'service.host.id' AFTER id;
+UPDATE service_state INNER JOIN service ON service.id = service_state.service_id SET service_state.host_id = service.host_id WHERE service_state.host_id IS NULL;
```

Comment:
Just tested this with MariaDB 10.6.4 and the first statement seems to fill the column with a 0x00...00 value, so the second statement affects no rows (filters for `host_id IS NULL` which is marked `NOT NULL`).